apps/api/v2/src/modules/organizations/controllers/memberships/organizations-membership.controller.ts

```diff
@@ -0,0 +1,124 @@
+import { CreateOrgMembershipOutput } from "@/modules/organizations/outputs/organization-membership/create-membership.output";
+import { DeleteOrgMembership } from "@/modules/organizations/outputs/organization-membership/delete-membership.output";
+import { GetAllOrgMemberships } from "@/modules/organizations/outputs/organization-membership/get-all-memberships.output";
+import { GetOrgMembership } from "@/modules/organizations/outputs/organization-membership/get-membership.output";
+import { OrgMembershipOutputDto } from "@/modules/organizations/outputs/organization-membership/membership.output";
+import { UpdateOrgMembership } from "@/modules/organizations/outputs/organization-membership/update-membership.output";
+import { OrganizationsMembershipService } from "@/modules/organizations/services/organizations-membership.service";
+import {
+  Controller,
+  UseGuards,
+  Get,
+  Param,
+  ParseIntPipe,
+  Query,
+  Delete,
+  Patch,
+  Post,
+  Body,
+  HttpCode,
+  HttpStatus,
+} from "@nestjs/common";
+import { ApiTags as DocsTags } from "@nestjs/swagger";
+import { plainToClass } from "class-transformer";
+
+import { SUCCESS_STATUS } from "@calcom/platform-constants";
+import { SkipTakePagination } from "@calcom/platform-types";
+
+@Controller({
+  path: "/v2/organizations/:orgId/memberships",
+  version: API_VERSIONS_VALUES,
+})
+@UseGuards(ApiAuthGuard, IsOrgGuard, RolesGuard)
+@DocsTags("Organizations Memberships")
+export class OrganizationsMembershipsController {
+  constructor(private organizationsMembershipService: OrganizationsMembershipService) {}
+
+  @Roles("ORG_ADMIN")
+  @Get("/")
+  async getAllMemberships(
+    @Param("orgId", ParseIntPipe) orgId: number,
+    @Query() queryParams: SkipTakePagination
+  ): Promise<GetAllOrgMemberships> {
+    const { skip, take } = queryParams;
+    const memberships = await this.organizationsMembershipService.getPaginatedOrgMemberships(
+      orgId,
+      skip ?? 0,
+      take ?? 250
+    );
+    return {
+      status: SUCCESS_STATUS,
+      data: memberships.map((membership) =>
+        plainToClass(OrgMembershipOutputDto, membership, { strategy: "excludeAll" })
+      ),
+    };
+  }
+
+  @Roles("ORG_ADMIN")
+  @Post("/")
+  async createMembership(
+    @Param("orgId", ParseIntPipe) orgId: number,
+    @Body() body: CreateOrgMembershipDto
+  ): Promise<CreateOrgMembershipOutput> {
+    const membership = await this.organizationsMembershipService.createOrgMembership(orgId, body);
+    return {
+      status: SUCCESS_STATUS,
+      data: plainToClass(OrgMembershipOutputDto, membership, { strategy: "excludeAll" }),
+    };
+  }
+
+  @Roles("ORG_ADMIN")
+  @UseGuards(IsMembershipInOrg)
+  @Get("/:membershipId")
+  async getUserSchedule(
+    @Param("orgId", ParseIntPipe) orgId: number,
+    @Param("membershipId", ParseIntPipe) membershipId: number
+  ): Promise<GetOrgMembership> {
+    const membership = await this.organizationsMembershipService.getOrgMembership(orgId, membershipId);
```

Comment:
the guard IsMembershipInOrg is already populating the request with the correct membership, use the decorator      @GetMembership instead